### PR TITLE
KAFKA-4738: Remove generic type of class ClientState

### PR DESCRIPTION
**Pull request by sharad-develop**
Remove generic type of class ClientState and generic T inTaskAssignor.

### CODE REVIEW

**Comment by sharad-develop** [comment]
retest this please

**Comment by sharad-develop** [comment]
KAFKA:4738 - Remove generic type of class ClientState

**Comment by sharad-develop** [comment]
This fixes - KAFKA:4738 - Remove generic type of class ClientState

**Comment by mjsax** [comment]
Please update PR title to `KAFKA-4738:  Remove generic type of class ClientState`

Also please resolve merge conflict. Otherwise, we cannot review.

**Comment by sharad-develop** [comment]
retest this please

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java [hunk 1]
```diff
@@ -137,20 +137,20 @@ private void assignTaskToClient(final Set<TaskId> assigned, final TaskId taskId,
     }
 
 
-    private ClientState<TaskId> findClient(final TaskId taskId,
+    private ClientState findClient(final TaskId taskId,
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java]: Nit: fix indention

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/ClientStateTest.java [hunk 2]
```diff
@@ -37,126 +38,126 @@ public void shouldHaveNotReachedCapacityWhenAssignedTasksLessThanCapacity() thro
...
         assertTrue(client.reachedCapacity());
     }
 
 
     @Test
     public void shouldAddActiveTasksToBothAssignedAndActive() throws Exception {
-        client.assign(1, true);
-        assertThat(client.activeTasks(), equalTo(Collections.singleton(1)));
-        assertThat(client.assignedTasks(), equalTo(Collections.singleton(1)));
+        client.assign(new TaskId(0, 1), true);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/ClientStateTest.java]: add a var `final TaskId tid = ...` (this should have been there for `int id = 1` already. Would improve the code base. Also for the other tests where it makes sense.

**Review by mjsax — CHANGES_REQUESTED** [review]
Only some minor comments.

**Comment by sharad-develop** [comment]
retest this please

**Comment by sharad-develop** [comment]
retest this please

**Comment by mjsax** [comment]
@sharad-develop It's ok of one test run fails -- no need to trigger a retest until all three pass. Just FYI.

**Review by mjsax — APPROVED** [review]
LGTM.
\cc @guozhangwang Call for final review and merging.

**Review by guozhangwang — APPROVED** [review]
Merged to trunk.

**Pull request closed without merging**